### DISCUSSION TITLE
Relative expression differences between v10 and older causing problems in MoBi

### DISCUSSION THREAD

**Discussion by RobinM92** [Open-Systems-Pharmacology/Forum#1151] [Questions & Problems]
Dear community,

I am trying to model a compound that binds to EGFR and undergoes TMDD. After including EGFR, based on the gene expression database, into PKSim (expressed interstitially, as there's no real tissue-specific apical cell membrane if I am not mistaken?) and adding binding parameters derived from in vitro, I exported the simulation to MoBi to create EGFR synthesis, degradation and internalisation reactions as is done for example in the ADC-tumor tutorial. However, to include the relative expression one should use the (relatively assigned) parameter 'Relative Expression Out.' (e.g. ..||EGFR|Relative expression out.) but this does not seem to be present. The only expression related parameter on that location is 'Relative expression', but this leads to problems with the paths and creates errors such as: 
![image](https://user-images.githubusercontent.com/30694339/155517449-7cb18736-de09-4fd2-ad53-1d9bcbe65ec3.png)

Looking closer, there seem to be several differences between the way protein expression is handled between v10 and v9 (as also stated in the release notes), so I was wondering if this could have caused this problem and what the 'new' way of handling relative expression in MoBi then would be? 

Thank you for your insights!

As a bonus, I tried to open the 2017 tutorials files in MoBi, but they also crash giving the following error, which might be related to the same issue:

Application:
MoBi® 10.0.122

could not execute query
[ SELECT this_.Id as id1_0_1_, this_.CommandId as commandid2_0_1_, this_.Discriminator as discriminator3_0_1_, this_.CommandInverseId as commandinverseid4_0_1_, this_.CommandType as commandtype5_0_1_, this_.ObjectType as objecttype6_0_1_, this_.Description as description7_0_1_, this_.ExtendedDescription as extendeddescription8_0_1_, this_.Visible as visible9_0_1_, this_.Comment as comment10_0_1_, this_.Sequence as sequence11_0_1_, this_.ParentId as parentid12_0_1_, commandmet2_.Id as id1_0_0_, commandmet2_.CommandId as commandid2_0_0_, commandmet2_.Discriminator as discriminator3_0_0_, commandmet2_.CommandInverseId as commandinverseid4_0_0_, commandmet2_.CommandType as commandtype5_0_0_, commandmet2_.ObjectType as objecttype6_0_0_, commandmet2_.Description as description7_0_0_, commandmet2_.ExtendedDescription as extendeddescription8_0_0_, commandmet2_.Visible as visible9_0_0_, commandmet2_.Comment as comment10_0_0_, commandmet2_.Sequence as sequence11_0_0_, commandmet2_.ParentId as parentid12_0_0_ FROM COMMANDS this_ left outer join COMMANDS commandmet2_ on this_.ParentId=commandmet2_.Id ]
[SQL: SELECT this_.Id as id1_0_1_, this_.CommandId as commandid2_0_1_, this_.Discriminator as discriminator3_0_1_, this_.CommandInverseId as commandinverseid4_0_1_, this_.CommandType as commandtype5_0_1_, this_.ObjectType as objecttype6_0_1_, this_.Description as description7_0_1_, this_.ExtendedDescription as extendeddescription8_0_1_, this_.Visible as visible9_0_1_, this_.Comment as comment10_0_1_, this_.Sequence as sequence11_0_1_, this_.ParentId as parentid12_0_1_, commandmet2_.Id as id1_0_0_, commandmet2_.CommandId as commandid2_0_0_, commandmet2_.Discriminator as discriminator3_0_0_, commandmet2_.CommandInverseId as commandinverseid4_0_0_, commandmet2_.CommandType as commandtype5_0_0_, commandmet2_.ObjectType as objecttype6_0_0_, commandmet2_.Description as description7_0_0_, commandmet2_.ExtendedDescription as extendeddescription8_0_0_, commandmet2_.Visible as visible9_0_0_, commandmet2_.Comment as comment10_0_0_, commandmet2_.Sequence as sequence11_0_0_, commandmet2_.ParentId as parentid12_0_0_ FROM COMMANDS this_ left outer join COMMANDS commandmet2_ on this_.ParentId=commandmet2_.Id]
-----------------------------------------------------
file is not a database
file is not a database

Stack trace:
```
at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String& strRemain)
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at NHibernate.AdoNet.AbstractBatcher.ExecuteReader(DbCommand cmd)
   at NHibernate.Loader.Loader.GetResultSet(DbCommand st, QueryParameters queryParameters, ISessionImplementor session, IResultTransformer forcedResultTransformer)
   at NHibernate.Loader.Loader.DoQuery(ISessionImplementor session, QueryParameters queryParameters, Boolean returnProxies, IResultTransformer forcedResultTransformer)
   at NHibernate.Loader.Loader.DoQueryAndInitializeNonLazyCollections(ISessionImplementor session, QueryParameters queryParameters, Boolean returnProxies, IResultTransformer forcedResultTransformer)
   at NHibernate.Loader.Loader.DoList(ISessionImplementor session, QueryParameters queryParameters, IResultTransformer forcedResultTransformer)
 --- End of inner exception stack trace ---
   at NHibernate.Loader.Loader.DoList(ISessionImplementor session, QueryParameters queryParameters, IResultTransformer forcedResultTransformer)
   at NHibernate.Impl.SessionImpl.List(CriteriaImpl criteria, IList results)
   at NHibernate.Impl.CriteriaImpl.List(IList results)
   at NHibernate.Impl.CriteriaImpl.List[T]()
   at OSPSuite.Infrastructure.Serialization.ORM.MetaData.CommandMetaDataRepository.LoadFromSession(ISession session)
   at OSPSuite.Infrastructure.Serialization.ORM.MetaData.HistoryItemMetaDataRepository.All(ISession session)
   at OSPSuite.Infrastructure.Serialization.ORM.History.HistoryManagerPersistor.allHistoryItemMetaData(ISession session)
   at OSPSuite.Infrastructure.Serialization.ORM.History.HistoryManagerPersistor.Load(ISession session)
   at MoBi.Core.Serialization.ORM.ContextPersistor.Load(IMoBiContext context, String projectFullPath)
   at MoBi.Presentation.Tasks.SerializationTask.<>c__DisplayClass10_0.<LoadProject>b__0()
   at OSPSuite.Presentation.Services.HeavyWorkManager.doWork()
   at OSPSuite.Presentation.Services.HeavyWorkManager.<.ctor>b__9_1(Object o, DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.OnDoWork(DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.WorkerThreadStart(Object argument)
```

**Comment by msevestre** [comment]
@RobinM92 
Parameter with out etc...where removed from version 10. So you can simply use the parameter called Relative Expression if you need to access it.
The error specifies that a parameter is not found . Your path does not seem to be correct. How did you define it?

as for the error, it looks like you are not opening the right file
`file is not a database`. Means that either you are opening a corrupted file, or a zipped file or sthg like this.
What is the URL of this file you are trying to open?

**Reply by RobinM92** [reply]
@msevestre 
Thank you for the quick reply! I used the relative path ..|EGFR_Array|Relative expression, where EGFR_Array is the name of the receptor. But perhaps I need to use a path relative to the synthesis process? I'll play around a bit, but good to know that 'Relative expression' can be used as such.

Regarding the file, I downloaded the mobi file from the repository here on github: https://github.com/Open-Systems-Pharmacology/ISOP-webinar---PK-Sim-MoBi-introduction/blob/master/Ex_1_ADC_Tox_Modeling_9.mbp3

**Reply by msevestre** [reply]
How did you download the file? I am pretty sure you downloaded the page associated with the file

**Reply by RobinM92** [reply]
In various ways, among them downloading the entire zipped folder and unzipping it and clicking 'view raw' to save. All generated mbp3 files, but indeed not all were the same size and the file generated from the 'view raw' opened in MoBi without errors. It seems however that there are some errors there regarding the tumor compartment, but I think I saw that here in the discussion forum already and those would not be relevant for the problem at hand, but it also means I cannot check if using 'Relative Expression' rathar than 'Relative expression out.' would work in this example file.

![image](https://user-images.githubusercontent.com/30694339/155531483-6bce8dbc-ea47-4336-93ac-e215c0b617ce.png)


So that brings me back to the path for the Relative Expression. Could it have to do something with the further splitting up into subcompartments for relative expression in v10? So I need to include them all formally rather than working with one relative path? Or should I change something in the MSV/PSV, perhaps MoBi has problems with tissues were there is no expression (NA instead of 0)?

E.g.

**Comment by RobinM92** [comment]
Dear all, 
Just to boil down the question to the essential:
Has anyone developed a model in MoBi in v10 using a protein binding partner or transporter with relative expressions from the database used in the synthesis process? If so, which path did you use to access the relative expression per tissue? 
Thank you!

**Reply by msevestre** [reply]
Can you attach your project?

**Reply by RobinM92** [reply]
[ProjectFILEForOSPforum.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/8140252/ProjectFILEForOSPforum.zip)


@msevestre Here you go, I removed observed data and such but the behaviour is the same; when you create a simulation it will not work because of the path resolution of relative expression.

Thank you so much for your help!

**Reply by RobinM92** [reply]
Hi @msevestre , were you able to have a look? Thanks so much for the effort!

**Reply by msevestre** [reply]
Hi @RobinM92 


`Unable to resolve path '..|EGFR_ARRAY|Relative expression' for entity 'EGFR_synthesis' with path 'S|Organism|VenousBlood|Plasma|EGFR_synthesis'`



I can reproduce the issue. It looks like you have no relative expression parameter (..|EGFR_ARRAY|Relative expression) defined for your Molecule EGFR_ARRAY used in the reaction `EGFR_synthesis` that is being constructed in Venous blood plasma (`S|Organism|VenousBlood|Plasma|EGFR_synthesis`)
And indeed, I cannot find any value defined in the PSV.  EIther create a value in your PSV or mark this value as "not present" in venous blood plasma 


I would suggest exporting a simple molecule from PK-Sim v10 and see how this is exported for plasma compartments (one value used by all parameters so this is a bit tricky)

**Reply by StephanSchaller** [reply]
I believe this may be related to: https://github.com/Open-Systems-Pharmacology/MoBi/issues/691#

**Reply by msevestre** [reply]
I don't think so. Molecule is defined as present but there is no value for rel exp in those organs .

**Reply by msevestre** [reply]
However having a tag mechanism could help solving the issue. But for a quick fix, I'd would try what I suggested above

**Reply by user1maria** [reply]
Dear @msevestre , I have the same Problem as RobinM92 and I don't understand your suggestion from above "EIther create a value in your PSV or mark this value as "not present" in venous blood plasma".
In my PSVs the Relative expressions for any subcompartments but intracellular, are simply not listed. How is it possible to create them?
Thanks

**Comment by PavelBal** [comment]
Stephan is right, this is related to that issue. The parameters

Fraction expressed interstitial
Fraction expressed intracellular
Relative expression

are not being created as they are NaN by default, and they are not extended in the PSV-BB. @RobinM92 can you upload or send me the project file? I will try to fix it.

**Reply by RobinM92** [reply]
Hi Pavel! I uploaded a version of the project in which Michael was also able to replicate the issue above, but here it is again: 
[ProjectFILEForOSPforum.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/8197271/ProjectFILEForOSPforum.zip)

Setting the Molecule EGFR_ARRAY to be 'not present' in the venous plasma indeed allows the simulation to be built, but now I am not sure if I am not missing anything else so it would be amazing indeed if you could have a look as well :)

**Reply by Salih-Benamara** [reply]
Hello, @RobinM92 please I have the same problem as you, I am trying to model the TMDD for infliximab and I have the same error, you manage to run the model? Are there any suggestions to make this work? . Thank You !

**Reply by RobinM92** [reply]
@PavelBal did you have a chance to have a look? For now I fixed it by not using the relative expression part, but of course that is not a great solution as the receptor will then converge to the same level in all tissues...

**Reply by PavelBal** [reply]
@RobinM92 Sorry, I did not have the time :/

You have to add these parameters for the new compartments manually. Take any other organ as a reference.

**Comment by PavelBal** [comment]
@RobinM92 Regarding the second issue with the parameters related to TwoPoresModel - see https://github.com/Open-Systems-Pharmacology/MoBi/issues/603

Yes, as a summary - all (large molecules) MoBi models developed with version below V10 are broken in V10 and require manual fixing...

**Reply by prvmalik** [reply]
likely re-estimation rather than manual fix if the targets are in more than 1 organ.

**Comment by Fan6687** [comment]
Hi, may I ask how much did you set for the reference concentration of EGFR? I didn't find the reference concentration value from literatures. Look forward to your reply. Thanks.

**Reply by RobinM92** [reply]
Hi @Fan6687 ,

I used two publications and the expression database to reverse engineer a reference concentration of EGFR:

1) The reference concentration of EGFR was calculated based on two approaches. For soluble EGFR, reference plasma concentrations of  30-60 ng/mL (0.27 – 0.55 nM) have been reported (Maramotti S, Paci M, Manzotti G, Rapicetta C, Gugnoni M, Galeone C, et al. Soluble epidermal growth factor receptors (sEGFRs) in cancer: Biological aspects and clinical relevance. International Journal of Molecular Sciences. ). Plasma concentrations are approximately 1% of the reference EGFR concentration as reported by the applied UNIGENE database ==> reference EGFR concentration in the body between 27 and 55 nM. 

2) Reported hepatic EGFR expression levels of 2 x 10^5 receptors per hepatocyte (Sorkin A, Duex JE. Quantitative Analysis of Endocytosis and Turnover of Epidermal Growth Factor (EGF) and EGF Receptor. Current Protocols in Cell Biology). Taking this information together with the amount of hepatocytes per mL liver (110 x 10^6, (Wilson ZE, Rostami-Hodjegan A, Burn JL, Tooley A, Boyle J, Ellis SW, et al. Inter-individual variability in levels of human microsomal protein and hepatocellularity per gram of liver. British Journal of Clinical Pharmacology 5)) leads to a hepatic EGFR concentration of 36.5 nM. Approximately  90% of reference EGFR is expressed in the liver according to the UNIGENE database ==> reference concentration of EGFR in the body is ~41 nM.  

The average of these values (27, 55, 41 nM) is also 41 nM, so this was deemed a good starting point

**Reply by Fan6687** [reply]
Many thanks for your so detailed and prompt reply! May I ask another two questions as following:
1) How did you set the relative expression value of EGFR targets in human tumor if the EGFR reference concentration was set as 41 nM?
2) Have you built the PBPK/PD model of ADC which is composed of antibody (e.g, EGFR antibody) and toxic payload? if yes, may I ask how did you construct the tumor inhibition model? Since the EGFR ADC can inhibit tumor growth by two approaches, one is to block the downstream signal pathway by combine with the human EGFR target, the other is to kill tumor cell by releasing toxic payload. I am confusing how to set these two mechanistic ways. I would be very appreciated if you could share some solutions.
Look forward to your reply! Thanks again!!!

**Reply by RobinM92** [reply]
Hi @Fan6687 ,

In this particular model, no tumor was included. If I would include it (in MoBi), I would probably take some assumptions on expression based on the fact that it overexpresses EGFR; e.g. a 10-fold relative expression. But I would looking the literature as well to see if anything is known about receptor expression in egfr+ tumors and try to incorporate that information. 

If I am not mistaken, the ADC-tutorials that are available use an empirical tumor growth model (e.g. Simeoni model) and then allow the compound to work on the parameters of this model, informed by either preclinical data or even clinical observations if they are there. You could just link the ADC and antibody concentration in the tumor interstitial space using e.g. a emax model.
